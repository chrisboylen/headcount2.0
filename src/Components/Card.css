.card {
  /* border: solid 1px; */
  border-radius: 15px;
  box-shadow: 0px 0px 19px 1px rgba(0, 0, 0, .7);
  flex: 1 18em;
  list-style-type: none;
  margin: 0 2em 2em 2em;
  padding: 0 1em 1em 1em;
}

h2 {
  font-size: 2em;
  margin: .5em;
  text-align: center;
}

li {
  font-size: 1.2em;
  padding: 0 0 0 3em;
}

.stat {
  padding: 0 0 .5em 0;
}