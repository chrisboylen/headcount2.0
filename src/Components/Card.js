import React from 'react';
import PropTypes from 'prop-types';

const Card = ({ location, stats}) => {
  const displayStats = Object.entries(stats).map((stat, i) => {
    return <div key={ i }>{stat[0]} : {stat[1]}</div>
  })
  
  return (
    <article className="card">
      <h1>{location}</h1>
      <ul>{displayStats}</ul>
    </article>
  )
}

Card.propTypes = {
  location: PropTypes.string,
  stats: PropTypes.objectOf(PropTypes.number)
}

export default Card;
