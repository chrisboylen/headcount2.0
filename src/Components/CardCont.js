import React from 'react';
import Card from './Card';
import PropTypes from 'prop-types';

const CardCont = ({ data }) => {
  console.log(data)
  const districtCards = Object.values(data).map((district, i) => {
    return <Card  location={ district.location }
                  stats={ district.stats }
                  key={ i }
    />
  })

  return (
    <section className="card-cont">
      { districtCards }
    </section>
  )
}

CardCont.propTypes = {
  data: PropTypes.shape({
    location: PropTypes.string,
    stats: PropTypes.objectOf(PropTypes.number)
  })
}

export default CardCont;