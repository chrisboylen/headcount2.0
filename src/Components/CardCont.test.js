import React from 'react';
import { shallow } from 'enzyme';
import CardCont from './CardCont';
import Card from './Card';

describe('CARDCONT', () => {
  it('Should match snapshot', () => {
    const wrapper = shallow(
      <CardCont data={[]} />);
    //   <Card 
    //     location='COLORADO'
    //     stats={{key: 1}}
    //     key={1}
    //   />
    // );

    expect(wrapper.html()).toMatchSnapshot();
  })

  it('Should invoke displayDistrictCards', () => {
    const mockData = [
      {
        location: 'COLORADO',
        stats: {
          2002: 1,
          2003: 6,
          2004: 8,
          2005: 0
        },
        id: 1
      },
      {
        location: 'BINGO',
        stats: {
          2002: 1,
          2007: 6,
          2004: 8,
          2009: 0
        },
        id: 2
      }
    ];

    const wrapper = shallow(
      <CardCont data={mockData} />);

    expect(wrapper.html()).toMatchSnapshot();
  })
})