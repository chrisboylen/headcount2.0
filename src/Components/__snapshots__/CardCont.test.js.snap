// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CARDCONT Should match snapshot when there is an empty array 1`] = `"<article class=\\"card-cont\\"></article>"`;

exports[`CARDCONT Should match snapshot when there is data in the array 1`] = `"<article class=\\"card-cont\\"><ul class=\\"card  \\"><h2>COLORADO</h2><li><div class=\\"stat\\">2002 : 1</div><div class=\\"stat\\">2003 : 6</div><div class=\\"stat\\">2004 : 8</div><div class=\\"stat less-than\\">2005 : 0</div></li></ul><ul class=\\"card  \\"><h2>BINGO</h2><li><div class=\\"stat\\">2002 : 1</div><div class=\\"stat\\">2004 : 8</div><div class=\\"stat\\">2007 : 6</div><div class=\\"stat less-than\\">2009 : 0</div></li></ul></article>"`;
